crates/dht/build.rs

```diff
@@ -0,0 +1,32 @@
+use handlebars::Handlebars;
+use std::collections::HashMap;
+use std::io::BufRead;
+use std::path::Path;
+
+fn main() {
+    println!("cargo::rerun-if-changed=DOCS.md.hbs");
+    println!("cargo::rerun-if-changed=art/dht-graph.svg");
+    println!("cargo::rerun-if-changed=art/dht-circle.svg");
+
+    let mut reg = Handlebars::new();
+    reg.register_template_file("DOCS.md", "DOCS.md.hbs")
+        .expect("Failed to register template");
+
+    let mut data = HashMap::new();
+    data.insert("dht-graph", load_svg("art/dht-graph.svg"));
+    data.insert("dht-circle", load_svg("art/dht-circle.svg"));
+
+    let mut f =
+        std::fs::File::create("DOCS.md").expect("Failed to create DOCS.md");
+    reg.render_to_write("DOCS.md", &data, &mut f)
+        .expect("Failed to render template");
```

Comment:
Rather than writing a file in the source directory that has to be git ignored, it is canonical to write it to the `OUT_DIR` directory, so it ends up in the target folder.

Reply:
This is gone, but yes, you're right